Platform options (#998)
AvaloniaUI/avalonia-docs: docs/platform-specific-guides/android/android.md

With the above _Android_ development environment setup, you will be able to build _Android_ applications, and run them in a simulator on your platform.

## Configuring Android platform options

`AndroidPlatformOptions` controls how Avalonia renders on Android. Apply it by overriding `CustomizeAppBuilder` in your `AvaloniaAndroidApplication` subclass:

```csharp
protected override AppBuilder CustomizeAppBuilder(AppBuilder builder)
{
    return base.CustomizeAppBuilder(builder)
        .With(new AndroidPlatformOptions
        {
            RenderingMode = new[]
            {
                AndroidRenderingMode.Egl,
                AndroidRenderingMode.Software
            }
        });
}
```

### Rendering mode

> `RenderingMode` is an ordered list of graphics backends. Avalonia tries each one in turn and uses the first that initializes successfully, so the first entry has the highest priority. The default is `Egl`, then `Software`.

Reviewer: ```suggestion
`RenderingMode` is an ordered list of graphics backends. Avalonia tries each one in turn and uses the first that initializes successfully, so the first entry has the highest priority. The default is a two-item list of `Egl` first, `Software` second.
```

| Mode | Description |
|---|---|

> | `Egl` | GPU rendering through EGL (OpenGL ES). The default GPU backend. |

Reviewer: ```suggestion
| `Egl` | GPU rendering through EGL (OpenGL ES). Default GPU backend. |
```

| `Vulkan` | GPU rendering through Vulkan. |
| `Software` | CPU rendering into a framebuffer. |

> To support the widest range of devices, include `Software` as a fallback. If `RenderingMode` is empty, or none of the listed modes initialize, Avalonia throws an `InvalidOperationException`.

Reviewer: ```suggestion
To support the widest range of devices, include `Software` as a fallback.

`RenderingMode` must contain at least one mode. If it is empty, or none of the listed modes initialize, Avalonia throws an `InvalidOperationException`.
```
